### PR TITLE
ARROW-115 Changelog for 0.5.0

### PR DESCRIPTION
**Pull request by blink1073**

### CODE REVIEW

File: bindings/python/docs/source/changelog.rst [hunk 1]
```diff
@@ -1,6 +1,12 @@
 Changelog
 =========
 
+Changes in Version 0.5.0
+------------------------
+- Support auto-discovery of schemas.
+- Support PyArrow 9.0.
+- Improve Error Message for Lib ImportError.
```
Comment (ShaneHarvey) [inline bindings/python/docs/source/changelog.rst:8]: "Improve error message for lib ..."

File: bindings/python/docs/source/changelog.rst [hunk 2]
```diff
@@ -1,6 +1,12 @@
 Changelog
 =========
 
+Changes in Version 0.5.0
+------------------------
+- Support auto-discovery of schemas.
```
Comment (ShaneHarvey) [inline bindings/python/docs/source/changelog.rst:6]: Worth mentioning the schema param is now option on the various find/aggregate_*_all methods. Also should we mention that it works by inferring the schema from the _first_ document in the result set?

File: bindings/python/docs/source/changelog.rst (hunk 1 above)
Comment (blink1073) [reply bindings/python/docs/source/changelog.rst:8]: Done

File: bindings/python/docs/source/changelog.rst (hunk 2 above)
Comment (blink1073) [reply bindings/python/docs/source/changelog.rst:6]: Done

**Review by ShaneHarvey — APPROVED** [review]

**Pull request merged**